### PR TITLE
New users segmentation in newsletters

### PR DESCRIPTION
**Pull request by laurajaime**
In the newsletters, now you can select users to send it.

diba_2

### CODE REVIEW

File: app/overrides/decidim/admins/newsletter/select_recipients_to_deliver.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+Deface::Override.new(virtual_path: "decidim/admin/newsletters/select_recipients_to_deliver",
```
Comment (tramuntanal) [inline app/overrides/decidim/admins/newsletter/select_recipients_to_deliver.rb:3]: Can you add the hash of the original file?

File: app/views/decidim/admin/newsletters/_select_single_users.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<div class="card-section send-newsletter-to-select-users">
+  <div class="row">
+    <div class="grid-x grid-padding-x">
+      <div class="cell small-12" id="send_newsletter_to_select_users">
+        <div class="row column">
+          <%= f.check_box :send_to_selected_users, label: t(".send_to_selected_users") %>
+          <p class="help-text"><%= t(".select_users.help") %></p>
+          <%= f.select(
+              :selected_users_ids,
+              Decidim::User.all.map{ |user| ["#{user.name} (#{user.email})", user.id] },
```
Comment (tramuntanal) [inline app/views/decidim/admin/newsletters/_select_single_users.html.erb:10]: Only users from current Organization

File: config/locales/ca.yml [hunk 3]
```diff
@@ -15,6 +15,15 @@ ca:
     messages:
       not_found: Email invàlid
   decidim:
+    admin:
+      newsletters:
+        index:
+          selected_users: Usuaris seleccionats
+        select_single_users:
+          send_to_selected_users: Enviar als usuaris seleccionats
```
Comment (tramuntanal) [inline config/locales/ca.yml:23]:
```suggestion
          send_to_selected_users: Envia als usuaris seleccionats
```

File: .rubocop_ruby.yml [hunk 4]
```diff
@@ -1094,12 +1094,12 @@ Metrics/ModuleLength:
 
 # Avoid complex methods.
 Metrics/CyclomaticComplexity:
-  Max: 9
+  Max: 40
```
Comment (tramuntanal) [inline .rubocop_ruby.yml:1097]: we can change to 10 but 40 is way too much

File: .rubocop_ruby.yml [hunk 5]
```diff
@@ -1122,7 +1122,7 @@ Metrics/ParameterLists:
     - "decidim-core/lib/decidim/filter_form_builder.rb"
 
 Metrics/PerceivedComplexity:
-  Max: 10
+  Max: 40
```
Comment (tramuntanal) [inline .rubocop_ruby.yml:1125]: Don't change please

File: .rubocop_ruby.yml [hunk 6]
```diff
@@ -1229,6 +1229,9 @@ RSpec/LetSetup:
 RSpec/MessageSpies:
   Enabled: false
 
+RSpec/MultipleMemoizedHelpers:
+  Max: 35
```
Comment (tramuntanal) [inline .rubocop_ruby.yml:1233]: Please, disable if needed but don't change the default

File: app/controllers/newsletter_settings_controller.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class NewsletterSettingsController < ApplicationController
+  def show
+    user = Decidim::User.find(params[:user_id])
```
Comment (tramuntanal) [inline app/controllers/newsletter_settings_controller.rb:5]: Filter by current Organization

File: app/decorators/queries/admin/newsletter_recipients_decorator.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+          recipients
+        else
+          original_query
+        end
+      end
+
+      private
+
+      def selected_users
+        Decidim::User.where(id: @form.selected_users_ids)
```
Comment (tramuntanal) [inline app/decorators/queries/admin/newsletter_recipients_decorator.rb:26]: Filter by current Organization

**Review by tramuntanal — CHANGES_REQUESTED** [review]
Great work here @laurajaime ! :clap: 

I've left some User scoping related comments and some other small things.

File: app/controllers/newsletter_settings_controller.rb (hunk 7 above)
Comment (laurajaime) [reply app/controllers/newsletter_settings_controller.rb:5]: wow...thanks!!!

File: .rubocop_ruby.yml (hunk 6 above)
Comment (tramuntanal) [reply .rubocop_ruby.yml]: I meant to disable on each place it raises, not to disable the Cop

**Review by tramuntanal — APPROVED** [review]
Ok

**Pull request merged**